tests/autorag/data/qacreation/test_base_qacreation.py

```diff
@@ -102,11 +102,24 @@ def test_make_qa_with_existing_queries(qa_parquet_filepath):
     assert all((elem in query_df['query'].tolist()) for elem in qa_df['query'].tolist())
 
 
-def test_make_qa_with_existing_queries_persistent_client(chroma_persistent_client, qa_parquet_filepath):
+def test_make_qa_with_existing_qa_with_gen_gt(qa_parquet_filepath):
+    corpus_df = pd.read_parquet(os.path.join(resource_dir, "corpus_data_sample.parquet"), engine='pyarrow')
+    query_df = pd.read_parquet(os.path.join(resource_dir, "qa_data_sample.parquet"), engine='pyarrow')
+    qa_df = make_qa_with_existing_qa(
+        corpus_df, query_df, content_size=5, exist_gen_gt=True,
+        output_filepath=qa_parquet_filepath, upsert=True,
+    )
+    validate_qa_dataset(qa_df)
+    assert len(qa_df) == 5
+    assert all(len(elem) == 3 for elem in qa_df['retrieval_gt'].apply(lambda x: x[0]).tolist())
+    assert all((elem in query_df['query'].tolist()) for elem in qa_df['query'].tolist())
```

Comment:
I think you need to add test case for testing existing generation_gt is properly merged at output qa_df